include/util/static_rtree.hpp

```diff
@@ -441,6 +441,14 @@ class StaticRTree
                 // inspecting an actual road segment
                 auto &current_candidate = current_query_node.node.template get<CandidateSegment>();
 
+                // to allow returns of no-results if too restrictive filtering, this needs to be done here
+                // even though performance would indicate that we want to stop after adding the first candidate
+                if (terminate(results.size(), current_candidate))
```

Comment:
We could technically filter the candidate based on its distance, not only abort. We already have the two parts. The abort would have to be a bit more sophisticated and we would duplicate the distance computation, however.